synapse/storage/schema/main/delta/58/05cache_instance.sql.postgres

```diff
@@ -28,3 +28,8 @@ CREATE TABLE cache_invalidation_stream_by_instance (
 CREATE UNIQUE INDEX cache_invalidation_stream_by_instance_id ON cache_invalidation_stream_by_instance(stream_id);
 
 CREATE SEQUENCE cache_invalidation_stream_seq;
+
+-- The sequence needs to begin at 2 because a bunch of code assumes that
+-- get_next_id_txn will return values >= 2, cf this comment:
+-- https://github.com/matrix-org/synapse/blob/b93bd95e8ab64d27ae26841020f62ee61272a5f2/synapse/storage/util/id_generators.py#L344
```

Comment:
Can we detect when the stream position is `<= 0` somewhere and throw an error to find our mistakes?

Seems like that could be at the end of the constructor of the `MultiWriterIdGenerator` after we calculate the `_current_positions`

Reply:
I was kind of trying to avoid spending more time on this TBH :) I can add an error in the logs but since triggering should be awfully rare (only on server creation or enabling replication the first time) I don't expect that we would easily catch all the code.

Reply:
@MatMaul [chatted](https://matrix.to/#/!eCNHSAdKTkIOXUNJoC:element.io/$ZNb3CATYl4q0uiQHGD6tm1sscdzjEqOi02pNPwZd8WA?via=matrix.org&via=element.io) with me a bit about this point. The goal of this change is just so that others don't have to figure out all of this context again when they run into this pitfall with another table and their tests just don't work like [what spawned this in the first place](https://github.com/matrix-org/synapse/pull/13667#discussion_r958644135).

The real solution around this would be to fix the weird stream ID's starting at `2` constraint. Should just work starting from any number ideally 🤷 

Instead of failing the process, @MatMaul mentioned a possibility of automatically correct and fix the problem by incrementing the stream sequence ID. But we don't really know if other tables have the same `2` constraint beyond the [comment in the code](https://github.com/matrix-org/synapse/blob/b93bd95e8ab64d27ae26841020f62ee61272a5f2/synapse/storage/util/id_generators.py#L344-L346) and don't want to overfix the problem.

In any case, maintainers are happy to move this forward without ⏩